src/ARTED/modules/opt_variables.f90

```diff
@@ -204,6 +210,21 @@ subroutine init_for_padding
     end do
   end subroutine
 
+  subroutine init_projector
+    use global_variables
+    implicit none
+    integer :: i
+
+    NP = Nlma / NI
+    NPI = Nps * NI
```

Comment:
Related to above comments, `NPI` might be efficiently described as `NPI=sum(Mps)`.